### PR TITLE
Update to Minecraft 1.20.6

### PR DESCRIPTION
**Pull request by JRoy**
this compiles and probably works but there is some things still left to do

- [x] test banner itemdb
- [x] fix potion itemdb (see todos)
- [x] update item aliases

### CODE REVIEW

**Review by CyberFlameGO — APPROVED** [review]
Changes look good to me - any ETA on a merge?

**Review by luftwafflezz — APPROVED** [review]
Awesome

**Review by ImDarkLaw — COMMENTED** [review]
Overall, changes look good. I noticed, however, that the following exception is still being printed in console when using `/give`. Perhaps that'd be something to address (or wait for Paper to resolve) before merging?

```log
[00:27:06 ERROR]: [org.bukkit.craftbukkit.util.CraftMagicNumbers] null
com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown item 'minecraft:' at position 0: <--[HERE]

File: Essentials/src/main/java/com/earth2me/essentials/Enchantments.java [hunk 1]
```diff
@@ -59,106 +70,127 @@ public final class Enchantments {
...
-        try {
-            final Enchantment depthStrider = Enchantment.getByName("DEPTH_STRIDER");
-            if (depthStrider != null) {
-                ENCHANTMENTS.put("depthstrider", depthStrider);
-                ALIASENCHANTMENTS.put("depth", depthStrider);
-                ALIASENCHANTMENTS.put("strider", depthStrider);
-            }
-        } catch (final IllegalArgumentException ignored) {
-        }
+        ENCHANTMENTS.put("depthstrider", Enchantment.DEPTH_STRIDER);
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/Enchantments.java:191]: Breaking 1.7 support in 2024 smh

File: Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java [hunk 2]
```diff
@@ -251,7 +250,7 @@ public void addStringMeta(final CommandSource sender, final boolean allowUnsafe,
             meta.setTitle(title);
             stack.setItemMeta(meta);
         } else if (split.length > 1 && split[0].startsWith("page") && split[0].length() > 4 && MaterialUtil.isEditableBook(stack.getType()) && hasMetaPermission(sender, "page", false, true, ess)) {
-            final int page = NumberUtil.isInt(split[0].substring(4)) ? (Integer.parseInt(split[0].substring(4)) - 1) : 0;
+            final int page = NumberUtil.isInt(split[0].substring(4)) ? Integer.parseInt(split[0].substring(4)) - 1 : 0;
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java]: IMO I prefer the extra brackets for clarity, even if they're technically redundant

**Comment by luftwafflezz** [comment]
> ```[00:27:06 ERROR]: [org.bukkit.craftbukkit.util.CraftMagicNumbers] null
> com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown item 'minecraft:' at position 0: <--[HERE]
> ```

Correct me if im wrong, but im pretty sure that the `minecraft:` prefix for an item is only supported in the vanilla `/give` command. For example, `/give @s minecraft:torch 1` would work in vanilla, but with essentials it would be `/give torch 1`.

**Comment by JRoy** [comment]
> > ```[00:27:06 ERROR]: [org.bukkit.craftbukkit.util.CraftMagicNumbers] null
> > com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown item 'minecraft:' at position 0: <--[HERE]
> > ```
> 
> Correct me if im wrong, but im pretty sure that the `minecraft:` prefix for an item is only supported in the vanilla `/give` command. For example, `/give @s minecraft:torch 1` would work in vanilla, but with essentials it would be `/give torch 1`.

the exception is unrelated to this and it is a (harmless) bukkit bug, nothing essentials can quite control.

**Comment by LOOHP** [comment]
> > > ```
> > > com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown item 'minecraft:' at position 0: <--[HERE]
> > > ```
> > 
> > 
> > Correct me if im wrong, but im pretty sure that the `minecraft:` prefix for an item is only supported in the vanilla `/give` command. For example, `/give @s minecraft:torch 1` would work in vanilla, but with essentials it would be `/give torch 1`.
> 
> the exception is unrelated to this and it is a (harmless) bukkit bug, nothing essentials can quite control.

I've encountered this with my plugin when I used `Bukkit.getUnsafe().modifyItemStack`. Apparently, we have to include the item type in the `arguments` to avoid this error in 1.20.6.

For example, this prints the `CommandSyntaxException` error:
```java
Bukkit.getUnsafe().modifyItemStack(head, "[minecraft:profile={properties:[{name:\"textures\",value:\"" + base64 + "\"}]}]");
```

This does not:
```java
Bukkit.getUnsafe().modifyItemStack(head, "minecraft:player_head[minecraft:profile={properties:[{name:\"textures\",value:\"" + base64 + "\"}]}]");
```

Previously, in 1.20.4, we did not need to include the item type:
```java
Bukkit.getUnsafe().modifyItemStack(head, "{SkullOwner: {Properties: {textures: [{Value: \"" + base64 + "\"}]}}}");
```

Most likely this is already known by Essentials devs but I thought I'd leave it here if anyone else is looking up this problem.

**Comment by Budderman18** [comment]
im noticing some potions in kits do not work (all types instant damage pots, there might be more). They show up as "null" when trying to create it

**Comment by ImDarkLaw** [comment]
> im noticing some potions in kits do not work (all types instant damage pots, there might be more). They show up as "null" when trying to create it

The issue with potions in kits, including instant damage potions (potion of harming), has been addressed in the latest commits to the [mc/1.20.5 branch](https://github.com/EssentialsX/Essentials/commits/mc/1.20.5/). Ensure you have pulled the latest changes from this branch and rebuilt the plugin. This should resolve potions showing up as *null*.

**Comment by Budderman18** [comment]
> > im noticing some potions in kits do not work (all types instant damage pots, there might be more). They show up as "null" when trying to create it
> 
> The issue with potions in kits, including instant damage potions (potion of harming), has been addressed in the latest commits to the [mc/1.20.5 branch](https://github.com/EssentialsX/Essentials/commits/mc/1.20.5/). Ensure you have pulled the latest changes from this branch and rebuilt the plugin. This should resolve potions showing up as _null_.

i am on the latest version of that branch and it still occurs.

Server version: 1.20.6-R0.1-SNAPSHOT 1.20.6-89-d3ffa62 (MC: 1.20.6)
Brand version: Paper
EssentialsX version: 2.21.0-dev+91-6bbc4a6

**Comment by JRoy** [comment]
> > > im noticing some potions in kits do not work (all types instant damage pots, there might be more). They show up as "null" when trying to create it
> > 
> > The issue with potions in kits, including instant damage potions (potion of harming), has been addressed in the latest commits to the [mc/1.20.5 branch](https://github.com/EssentialsX/Essentials/commits/mc/1.20.5/). Ensure you have pulled the latest changes from this branch and rebuilt the plugin. This should resolve potions showing up as _null_.
> 
> i am on the latest version of that branch and it still occurs.
> 
> Server version: 1.20.6-R0.1-SNAPSHOT 1.20.6-89-d3ffa62 (MC: 1.20.6)
> Brand version: Paper
> EssentialsX version: 2.21.0-dev+91-6bbc4a6

when the server is stopped, delete the items.json file in the plugins/Essentials folder. When you start it, the issue should be resolved.

This will happen automatically once merged

File: Essentials/src/main/resources/messages.properties [hunk 3]
```diff
@@ -863,8 +863,10 @@ noLocationFound=<dark_red>No valid location found.
 noMail=<primary>You do not have any mail.
 noMailOther=<secondary>{0} <primary>does not have any mail.
 noMatchingPlayers=<primary>No matching players found.
+noMetaComponents=Data Components are not supported in this version of Bukkit. Please use JSON NBT metadata.
 noMetaFirework=<dark_red>You do not have permission to apply firework meta.
 noMetaJson=JSON Metadata is not supported in this version of Bukkit.
+noMetaNbtKill=JSON NBT metadata is no longer supported, you must manually convert your defined items to data components. You can use Paper's tool to convert here: https://docs.papermc.io/misc/tools/item-command-converter
```
Comment (mdcfe) [inline Essentials/src/main/resources/messages.properties]:
```suggestion
noMetaNbtKill=JSON NBT metadata is no longer supported. You must manually convert your defined items to data components. You can convert JSON NBT to data components here: https://docs.papermc.io/misc/tools/item-command-converter
```
Slightly more explicit

File: providers/BaseProviders/src/main/java/net/ess3/provider/providers/BaseBannerDataProvider.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
+package net.ess3.provider.providers;
+
+import net.ess3.provider.BannerDataProvider;
+import org.bukkit.DyeColor;
+import org.bukkit.inventory.ItemStack;
+
+public class BaseBannerDataProvider implements BannerDataProvider {
+    @Override
+    public DyeColor getBaseColor(ItemStack stack) {
```
Comment (mdcfe) [inline providers/BaseProviders/src/main/java/net/ess3/provider/providers/BaseBannerDataProvider.java:33]: Is there really no better way to do this than a massive switch case

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**